sql/telemetry_derived/experiment_search_aggregates_live_v1/view.sql.py

```diff
@@ -0,0 +1,149 @@
+       Unfortunately, BigQuery does not allow parameters in view definitions, so this script is a very thin
+       wrapper over the nearly complete sql query to fill in the cutoff date for using the pre-aggregated vs
+       live table for the view"""
+    query = textwrap.dedent(
+        """
+        CREATE
+        OR REPLACE VIEW `moz-fx-data-shared-prod.telemetry_derived.experiment_search_aggregates_live_v1` AS
+          WITH all_experiments_searches_live AS (
+            SELECT
+              submission_timestamp AS timestamp,
+              unnested_experiments,
+              unnested_ad_clicks,
+              unnested_search_with_ads,
+              unnested_search_counts
+            FROM
+              `moz-fx-data-shared-prod.telemetry_live.main_v4`,
+            UNNEST(
+              ARRAY(SELECT AS STRUCT key, value.branch AS value FROM UNNEST(environment.experiments))
+            ) AS unnested_experiments,
+            UNNEST(payload.processes.parent.keyed_scalars.browser_search_ad_clicks) AS unnested_ad_clicks,
+            UNNEST(
+              payload.processes.parent.keyed_scalars.browser_search_with_ads
+            ) AS unnested_search_with_ads,
+            UNNEST(
+              ARRAY(
+                SELECT AS STRUCT
+                  SUBSTR(_key, 0, pos - 2) AS engine,
+                  SUBSTR(_key, pos) AS source,
+                  udf.extract_histogram_sum(value) AS `count`
+                FROM
+                  UNNEST(payload.keyed_histograms.search_counts),
+                  UNNEST([REPLACE(key, 'in-content.', 'in-content:')]) AS _key,
+                  UNNEST([LENGTH(REGEXP_EXTRACT(_key, '.+[.].'))]) AS pos
+              )
+            ) AS unnested_search_counts
+            WHERE
+              date(submission_timestamp) > '{submission_date}'
+              AND ARRAY_LENGTH(environment.experiments) > 0
+          ),
+          live AS (
+            SELECT
+              unnested_experiments.key AS experiment,
+              unnested_experiments.value AS branch,
+              TIMESTAMP_ADD(
+                TIMESTAMP_TRUNC(`timestamp`, HOUR),
+                -- Aggregates event counts over 5-minute intervals
+                INTERVAL(DIV(EXTRACT(MINUTE FROM `timestamp`), 5) * 5) MINUTE
+              ) AS window_start,
+              TIMESTAMP_ADD(
+                TIMESTAMP_TRUNC(`timestamp`, HOUR),
+                INTERVAL((DIV(EXTRACT(MINUTE FROM `timestamp`), 5) + 1) * 5) MINUTE
+              ) AS window_end,
+              SUM(unnested_ad_clicks.value) AS ad_clicks_count,
+              SUM(unnested_search_with_ads.value) AS search_with_ads_count,
+              SUM(unnested_search_counts.count) AS search_count
+            FROM
+              all_experiments_searches_live
+            GROUP BY
+              experiment,
+              branch,
+              window_start,
+              window_end
```

Comment:
rather than repeat this logic, I think it would be better to store this aggregation as a view that is used both here and when generating `experiment_search_aggregates_v1`

Reply:
There are some small differences between the logic here and in `experiment_search_aggregates_v1` that make this quite tricky:
 
This view queries all entries from the live table that haven't been written to `experiment_search_aggregates_v1` by doing `date(submission_timestamp) > '{submission_date}'`.

For the `experiment_search_aggregates_v1` we'd need to change this to `date(submission_timestamp) = '{submission_date}'` if we'd query from the live table. Or we leave this `WHERE` statement but would need to query from `telemetry_stable.main_v4`.

Reply:
moved to #1239